vignettes/BiocParallel_BatchtoolsParam.Rmd

```diff
@@ -0,0 +1,229 @@
+    number_sections: true
+    toc: yes
+    toc_depth: 4
+---
+ 
+# Introduction
+
+The class is an interface to `batchtools` the package from within `BiocParallel`. This aims to
+replace BatchjobsParam as `BiocParallel`'s class for computing on a high performance
+cluster such as SGE, TORQUE, LSF, SLURM, OpenLava.
+
+# Quick start
+
+This example demonstrates the easiest way to launch a 100000 jobs using
+batchtools. The first step involves creating a `BatchtoolsParam` class. You can compute
+using 'bplapply' and then the result is stored.
+
+
+```{r intro}
+library(BiocParallel)
+
+## Pi approximation
+piApprox <- function(n) {
+    nums <- matrix(runif(2 * n), ncol = 2)
+    d <- sqrt(nums[, 1]^2 + nums[, 2]^2)
+    4 * mean(d <= 1)
+}
+
+piApprox(1000)
+
+## Apply piApprox over
+param <- BatchtoolsParam()
+result <- bplapply(rep(10e5, 10), piApprox, BPPARAM=param)
+mean(unlist(result))
+```
+
+# *BatchtoolsParam* interface
+
+The interface allows you to replace the BatchJobsParam interface, and
+allow more intuitive usage of your high performance cluster with `BiocParallel`.
+
+The `BatchtoolsParam` class allows the user to specify many arguments to customize their
+jobs. Applicable to clusters with formal schedulers.
+
+-   `workers`The number of workers used by the job.
+
+-   `cluster` We currently support, SGE, SLURM, LSF, TORQUE and OpenLava. The
+    'cluster' argument is supported only if the R environment knows how
+    to find the job scheduler. Each cluster type uses a template to pass
+    the job to the scheduler. If the template is not given we use the
+    default templates as given in the 'batchtools' package. The cluster
+    can be accessed by 'bpbackend(param)'.
+
+-   `registryargs` The 'registryargs' argument takes a list of arguments to create a
+    new job registry for you `BatchtoolsParam`. The job registry is a data.table which
+    stores all the required information to process your jobs. The
+    arguments we support for registryargs are:
+    
+    + `file.dir` Path where all files of the registry are
+       saved. Note that some templates do not handle relative paths
+       well. If nothing is given, a temporary directory will be used
+       in your current working directory.
+        
+     + `work.dir` Working directory for R process for
+       running jobs.
+       
+     + `packages` Packages that will be loaded on each node.
+
+     + `namespaces` Namespaces that will be loaded on each
+        node.
+
+     + 'source` Files that are sourced before executing a
+        job.
+
+     + `load` Files that are loaded before executing a job.
+     
+    ```{r}
+        registryargs <- batchtoolsRegistryargs(
+        file.dir = "mytempreg",
+        work.dir = getwd(),
+        packages = character(0L),
+        namespaces = character(0L),
+        source = character(0L),
+        load = character(0L)
+        )
+        param <- BatchtoolsParam(registryargs = registryargs)
+        param
```

Comment:
To match the same spacing on the original vignette, please change the spacing from file.dir down to param to match this screenshot: 
<img width="528" alt="Screen Shot 2022-09-14 at 12 42 51 PM" src="https://user-images.githubusercontent.com/105299892/190213493-7026a8d1-fda6-4ecb-908d-2ccd1df57cc2.png">